cirq/experiments/fidelity_estimation.py

```diff
@@ -372,3 +394,225 @@ def least_squares_xeb_fidelity_from_probabilities(
     return least_squares_xeb_fidelity_from_expectations(
         measured_expectations, exact_expectations, uniform_expectations
     )
+
+
+@dataclass(frozen=True)
+class _Sample2qXEBTask:
+    """Helper contained for grouping a circuit to be sampled."""
+
+    cycle_depth: int
+    circuit_i: int
+    prepared_circuit: 'cirq.Circuit'
+
+
+class _SampleInBatches:
+    def __init__(self, sampler: 'cirq.Sampler', repetitions: int):
+        """This closure will execute a list of `tasks` with one call to
+        `run_batch` on the provided sampler for a given number of repetitions."""
+        self.sampler = sampler
+        self.repetitions = repetitions
+
+    def __call__(self, tasks: List[_Sample2qXEBTask]):
+        prepared_circuits = [task.prepared_circuit for task in tasks]
+        results = self.sampler.run_batch(prepared_circuits, repetitions=self.repetitions)
+        assert len(results) == len(tasks)
+        records = []
+        for task, n_result in zip(tasks, results):
+            assert len(n_result) == 1, len(n_result)
+            result = n_result[0]  # sweep nesting
+            sampled_inds = result.data.values[:, 0]
+            sampled_probs = np.bincount(sampled_inds, minlength=2 ** 2) / len(sampled_inds)
+
+            records += [
+                {
+                    'circuit_i': task.circuit_i,
+                    'cycle_depth': task.cycle_depth,
+                    'sampled_probs': sampled_probs,
+                }
+            ]
+        return records
+
+
+def sample_2q_xeb_circuits(
+    sampler: 'cirq.Sampler',
+    circuits: Sequence['cirq.Circuit'],
+    cycle_depths: Sequence[int],
+    repetitions: int = 10_000,
+    batch_size: int = 9,
```

Comment:
All the samplers support batching because of the fallback implementation of just running things in a for loop. I think the default should definitely be _with_ batching for these. You can always set it to `1`. And if there's a bug where it errors out, that's a failure of a sampler following the `Sampler` interface

Reply:
Alright.